tests/components/remove_duplicates_side_panel.test.ts

```diff
@@ -0,0 +1,174 @@
+    await click(fixture.querySelector(selectors.checkBoxHasHeaderRow)!);
+
+    // unselect column B
+    const checkBoxColumnB = fixture.querySelectorAll(selectors.checkBoxColumnsToAnalyzeInput)[2]; // checkBox[0] correspond to " Select all "
+    await click(checkBoxColumnB);
+
+    await click(fixture.querySelector(selectors.removeDuplicateButton)!);
+    expect(getEvaluatedCell(model, "B1").value).toBe("B1");
+    expect(getEvaluatedCell(model, "B2").value).toBe("B2");
+    expect(getEvaluatedCell(model, "B3").value).toBe("B3");
+    expect(getEvaluatedCell(model, "B4").value).toBe("B4"); // Because GOD != DOG
+    expect(getEvaluatedCell(model, "B5").value).toBe("");
+  });
+
+  test("checkboxes columns evolve with correct state ", async () => {
+    setCellContent(model, "A1", "Atchoum");
+    setSelection(model, ["A1:B2"]);
+    await nextTick();
+
+    let checkBoxes = fixture.querySelectorAll(selectors.checkBoxColumnsToAnalyzeInput);
+    let checkBoxAll = checkBoxes[0];
+    let checkBoxA = checkBoxes[1];
+    let checkBoxB = checkBoxes[2];
+
+    // at the beginning --> expect all checkbox to be selected
+    expect((checkBoxAll as HTMLInputElement).checked).toBe(true);
+    expect((checkBoxA as HTMLInputElement).checked).toBe(true);
+    expect((checkBoxB as HTMLInputElement).checked).toBe(true);
+
+    // unselect "select all" --> all checkbox should be unselected
+    await click(checkBoxAll);
+    await nextTick();
+    expect((checkBoxAll as HTMLInputElement).checked).toBe(false);
+    expect((checkBoxA as HTMLInputElement).checked).toBe(false);
+    expect((checkBoxB as HTMLInputElement).checked).toBe(false);
+
+    // select A --> only A should be selected
+    await click(checkBoxA);
+    await nextTick();
+    expect((checkBoxAll as HTMLInputElement).checked).toBe(false);
+    expect((checkBoxA as HTMLInputElement).checked).toBe(true);
+    expect((checkBoxB as HTMLInputElement).checked).toBe(false);
+
+    // select B --> "select all" become selected because all checkbox are selected
+    await click(checkBoxB);
+    await nextTick();
+    expect((checkBoxAll as HTMLInputElement).checked).toBe(true);
+    expect((checkBoxA as HTMLInputElement).checked).toBe(true);
+    expect((checkBoxB as HTMLInputElement).checked).toBe(true);
+
+    // unselect A --> "select all" become unselected, still B selected
+    await click(checkBoxA);
+    await nextTick();
+    expect((checkBoxAll as HTMLInputElement).checked).toBe(false);
+    expect((checkBoxA as HTMLInputElement).checked).toBe(false);
+    expect((checkBoxB as HTMLInputElement).checked).toBe(true);
+  });
+
+  test("if no content selected --> display error message and disable ", async () => {
+    setSelection(model, ["A1:B2"]);
+    await nextTick();
+    const errors = fixture.querySelectorAll(selectors.sidePanelErrors);
+    expect(errors.length).toBe(1);
+    expect(errors[0].textContent).toBe("Please select a range of cells containing values.");
```

Comment:
Use the constant in RemoveDuplicateTerms

Reply:
IMO, Don't really agree with that. In principle, we shouldn't use the way we implement a behavior to test that behavior.

So yes, here we would just reuse a constant, but I'd rather not have to worry about whether or not I can afford to reuse what has been defined in code.

To be sure never to do anything stupid, I prefer to avoid this practice as much as possible.